docs/Controllers.md

```diff
@@ -0,0 +1,87 @@
+The Controller SHOULD also refresh the endpoints of the relevant Resources to ensure the Controller is accurately reflecting the current state of the API.
+
+## Registry Service Discovery	
+
+In order to locate the Registry, the Controller SHOULD support all of the following: 
+* Unicast DNS Service Discovery (DNS-SD).
+* mDNS.
+* Direct configuration with the location of a preferred Registry.
+
+The Controller SHOULD offer unicast DNS-SD as the default mechanism. 
+
+Controllers SHOULD observe and interpret all of the TXT records returned with the DNS Service Discovery responses according to the requirements for Query API Clients specified in the [API Paths section of the APIs document](APIs.md#api-paths) in this specification.
+
+## Query API
+The Controller MUST be capable of using the Registry's IS-04 Query API to discover any registered resource, including Node, Device,  Source, Flow, Sender, and Receiver,
+as described in the [APIs section](README.md#apis) of this specification.
+
+The Controller MUST use the Registry�s IS-04 Query API either via the REST API or by requesting WebSocket subscriptions.
+
+When using the Query API, basic queries SHOULD be used (and advanced query language where available) to cut down on the volume of resources returned to the Controller.	
+
+Controllers SHOULD adhere to the version downgrade requirements for Query API Clients specified in the [Requirements for Query API Clients section of the Upgrade Path document](Upgrade%20Path.md#requirements-for-query-api-clients) in this specification.
+
+## Pagination
+In large systems, API resources rely upon pagination in order to return high volumes of data.
+This can be both hard to keep track of for a client, and require a large number of requests in order to scan the entire data set.
+
+For this reason it is recommended to use the RESTful QUERY API for debug and development purposes only.
+
+If using the RESTful Query API rather than WebSockets, Pagination requirements MUST be implemented as specified in the [Pagination section of the APIs: Query Parameters document](APIs%20-%20Query%20Parameters.md#pagination) in this specification.
+	
+## WebSockets & Subscriptions	
+Where a WebSocket or other subscription based mechanism is provided for Controller usage, it is strongly recommended that Controllers make use of this and do not use the API resources directly.
+
+If a WebSocket connection fails, then an attempt to reconnect to the WebSocket SHOULD NOT be attempted. Instead, a new subscription SHOULD be created with this API or a different one if required. If all available APIs return errors, an exponential backoff algorithm SHOULD be used when retrying until a success code is returned.
+
+## Dynamic Update of Resources
+The Controller MUST be capable of using the Registry's IS-04 Query API to discover and dynamically update the state of any registered resource, including Node, Device,  Source, Flow, Sender, and Receiver.
```

Comment:
I take "IS-04 Query API" to mean the HTTP API. And yet the above text strongly recommends against using it...

Reply:
I see a double space before `Source`.

Reply:
Good questions, not sure.

Reply:
The [Behaviour: Querying](https://github.com/AMWA-TV/is-04/blob/v1.3.x/docs/Behaviour%20-%20Querying.md) document says (my emphasis): 

> The Query API provides a read-only interface to the Nodes and their sub-resources which are currently held in the registry. Monitoring of these resources can be performed using the **HTTP API**, or using a **WebSocket connection** (RFC 6455), which is suggested for most applications.

So that appears to indicate that Query API is the superset, with HTTP API and WebSocket subscriptions as constituent parts.

Reply:
Agreed. During review let's ensure consistent usage of "HTTP API" where that is specifically intended.